SPECS/msft-golang/msft-golang.spec

```diff
@@ -0,0 +1,229 @@
+Source1:        https://dl.google.com/go/go1.4-bootstrap-20171003.tar.gz
+Patch0:         go14_bootstrap_aarch64.patch
+Obsoletes:      %{name} < %{version}
+Provides:       %{name} = %{version}
+Conflicts:      go
+Conflicts:      golang
+
+%description
+Go is an open source programming language that makes it easy to build simple, reliable, and efficient software.
+
+%prep
+# Setup go 1.4 bootstrap source
+tar xf %{SOURCE1} --no-same-owner
+patch -Np1 --ignore-whitespace < %{PATCH0}
+mv -v go go-bootstrap
+
+%setup -q -n go
+
+%build
+# Build go 1.4 bootstrap
+pushd %{_topdir}/BUILD/go-bootstrap/src
+CGO_ENABLED=0 ./make.bash
+popd
+mv -v %{_topdir}/BUILD/go-bootstrap %{_libdir}/golang
+export GOROOT=%{_libdir}/golang
+
+# Build current go version
+export GOHOSTOS=linux
+export GOHOSTARCH=%{gohostarch}
+export GOROOT_BOOTSTRAP=%{goroot}
+
+export GOROOT="`pwd`"
+export GOPATH=%{gopath}
+export GOROOT_FINAL=%{_bindir}/go
+rm -f  %{gopath}/src/runtime/*.c
+# Remove the explicit version once `microsoft/go#262` is resolved.
+echo 'go%{version}-1-fips' > VERSION
+pushd src
+./make.bash --no-clean
+popd
+
+%install
+
+mkdir -p %{buildroot}%{_bindir}
+mkdir -p %{buildroot}%{goroot}
+
+cp -R api bin doc lib pkg src misc VERSION %{buildroot}%{goroot}
+
+# remove the unnecessary zoneinfo file (Go will always use the system one first)
+rm -rfv %{buildroot}%{goroot}/lib/time
+
+# remove the doc Makefile
+rm -rfv %{buildroot}%{goroot}/doc/Makefile
+
+# put binaries to bindir, linked to the arch we're building,
+# leave the arch independent pieces in %{goroot}
+mkdir -p %{buildroot}%{goroot}/bin/linux_%{gohostarch}
+ln -sfv ../go %{buildroot}%{goroot}/bin/linux_%{gohostarch}/go
+ln -sfv ../gofmt %{buildroot}%{goroot}/bin/linux_%{gohostarch}/gofmt
+ln -sfv %{goroot}/bin/gofmt %{buildroot}%{_bindir}/gofmt
+ln -sfv %{goroot}/bin/go %{buildroot}%{_bindir}/go
+
+# ensure these exist and are owned
+mkdir -p %{buildroot}%{gopath}/src/github.com/
+mkdir -p %{buildroot}%{gopath}/src/bitbucket.org/
+mkdir -p %{buildroot}%{gopath}/src/code.google.com/p/
+
+install -vdm755 %{buildroot}%{_sysconfdir}/profile.d
+cat >> %{buildroot}%{_sysconfdir}/profile.d/go-exports.sh <<- "EOF"
+export GOROOT=%{goroot}
+export GOPATH=%{_datadir}/gocode
+export GOHOSTOS=linux
+export GOHOSTARCH=%{gohostarch}
+export GOOS=linux
+EOF
+
+%post -p /sbin/ldconfig
+%postun
+/sbin/ldconfig
+if [ $1 -eq 0 ]; then
+  #This is uninstall
+  rm %{_sysconfdir}/profile.d/go-exports.sh
+  rm -rf /opt/go
+  exit 0
+fi
+
+%files
+%defattr(-,root,root)
+%license LICENSE
+%exclude %{goroot}/src/*.rc
+%exclude %{goroot}/include/plan9
+%{_sysconfdir}/profile.d/go-exports.sh
+%{goroot}/*
+%{gopath}/src
+%exclude %{goroot}/src/pkg/debug/dwarf/testdata
+%exclude %{goroot}/src/pkg/debug/elf/testdata
+%{_bindir}/*
+
+%changelog
+* Thu May 05 2022 Muhammad Falak <mwani@microsoft.com> - 1.18.1-1
+- Switch to `microsoft/go` for a fips compliant version of go
```

Comment:
Why not just update the current Golang spec and say we're switching to Microsoft sources? Also, if we need to keep both versions, I wouldn't copy the changelog, since this is not the history for this file and if anything, I'd keep the attribution to Photon, since we're still using their way of building the package.

Reply:
To paint a better picture let me update the PR description a little more.
We had an internal agreement that in order to switch to `microsoft/go` from `golang/go` we should do a phased approach. We first should ship `microsoft/go` as an alternate package `msft-golang` and let both packages co-exist for sometime. Once we are confident, we can remove the upstream `golang.spec` and rename this current one to `golang`. 
This is one of the reasons why we did not modify the changelog entries.

Reply:
Thanks, Falak, makes sense.